models/intermediate/marketo__lead_calendar_spine.sql

```diff
@@ -1,9 +1,11 @@
 {{
```

Comment:
This model fails on my local runs due to the query exceeding resource limits. Do you also have the same issue?

Reply:
I think we should consider adding a `date_range_start` variable to the `marketo__calendar_spine` model. I believe this is why it fails and it is pretty restricting that we will **always** create a calendar spine looking back to 2016 (which I don't think Marketo users realistically need).

Let's leverage how we have done this in the past to allow this date range to be dynamically set by users if they wish to adjust it. See how we documented it in our [dbt_mixpanel](https://github.com/fivetran/dbt_mixpanel#event-date-range) package and where it was [declared](https://github.com/fivetran/dbt_mixpanel/blob/main/models/staging/stg_mixpanel__user_event_date_spine.sql#L25).

Reply:
I incorporated some of the logic from [this model](https://github.com/fivetran/dbt_zendesk/blob/d52992831530255366bd2ded70c16efb06265587/models/utils/int_zendesk__calendar_spine.sql) to make the start date either 
1. default the earliest date on record or 
2. can be set by a var.